### PR TITLE
Update Vpc object in store on change in content and improve logging.

### PR DESCRIPTION
**Pull request by archanapholla**

### CODE REVIEW

File: pkg/controllers/inventory/inventory.go [hunk 1]
```diff
@@ -47,36 +48,46 @@ func InitInventory() *Inventory {
 
 // BuildVpcCache using vpc list from cloud, update vpc cache(with vpcs applicable for the current cloud account).
 func (inventory *Inventory) BuildVpcCache(vpcMap map[string]*runtimev1alpha1.Vpc, namespacedName *types.NamespacedName) error {
```
Comment (reachjainrahul) [inline pkg/controllers/inventory/inventory.go:50]: nit: discoveredVpcMap

File: pkg/controllers/inventory/inventory.go [hunk 2]
```diff
@@ -47,36 +48,46 @@ func InitInventory() *Inventory {
...
 		vpc := i.(*runtimev1alpha1.Vpc)
 		if _, found := vpcMap[vpc.Status.Id]; !found {
-			inventory.log.V(1).Info("Deleting vpc from vpc cache", "vpc id", vpc.Status.Id, "account",
-				namespacedName.String())
 			if err := inventory.vpcStore.Delete(fmt.Sprintf("%v/%v-%v", vpc.Namespace, vpc.Labels[common.VpcLabelAccountName],
 				vpc.Status.Id)); err != nil {
 				inventory.log.Error(err, "failed to delete vpc from vpc cache", "vpc id", vpc.Status.Id, "account",
 					namespacedName.String())
 			}
+			numVpcsToDelete++
```
Comment (reachjainrahul) [inline pkg/controllers/inventory/inventory.go:63]: else { }. ++ for success case only

File: pkg/controllers/inventory/inventory.go [hunk 3]
```diff
@@ -47,36 +48,46 @@ func InitInventory() *Inventory {
...
-		if _, found, _ := inventory.vpcStore.Get(key); !found {
-			err = inventory.vpcStore.Create(v)
+		key := fmt.Sprintf("%v/%v-%v", discoveredVpc.Namespace, discoveredVpc.Labels[common.VpcLabelAccountName], discoveredVpc.Status.Id)
+		if cachedObj, found, _ := inventory.vpcStore.Get(key); !found {
+			err = inventory.vpcStore.Create(discoveredVpc)
+			numVpcsToAdd++
 		} else {
-			err = inventory.vpcStore.Update(v)
+			cachedVpc := cachedObj.(*runtimev1alpha1.Vpc)
+			if !utils.AreDiscoveredFieldsSameVpcStatus(cachedVpc.Status, discoveredVpc.Status) {
```
Comment (reachjainrahul) [inline pkg/controllers/inventory/inventory.go:75]: Doesnt deepequal compare work??

**Review by Anandkumar26 — COMMENTED** [review]
Please remove the file https://github.com/antrea-io/nephe/blob/main/pkg/apiserver/webhook/unit-test.log from the repository in this PR.

File: pkg/controllers/inventory/inventory.go (hunk 3 above)
Comment (archanapholla) [reply pkg/controllers/inventory/inventory.go]: It works, I had thought that it wouldn't work for nested objects like slices, I have updated the code. Also, tested.

File: pkg/controllers/inventory/inventory.go (hunk 2 above)
Comment (archanapholla) [reply pkg/controllers/inventory/inventory.go]: yes

File: pkg/controllers/inventory/inventory.go (hunk 1 above)
Comment (archanapholla) [reply pkg/controllers/inventory/inventory.go]: done

**Review by reachjainrahul — APPROVED** [review]
/LGTM

File: pkg/controllers/inventory/inventory.go [hunk 4]
```diff
@@ -46,37 +47,52 @@ func InitInventory() *Inventory {
...
+		if cachedObj, found, _ := inventory.vpcStore.Get(key); !found {
+			err = inventory.vpcStore.Create(discoveredVpc)
+			if err == nil {
+				numVpcsToAdd++
+			}
 		} else {
-			err = inventory.vpcStore.Update(v)
+			cachedVpc := cachedObj.(*runtimev1alpha1.Vpc)
+			if !reflect.DeepEqual(cachedVpc.Status, discoveredVpc.Status) {
+				err = inventory.vpcStore.Update(discoveredVpc)
```
Comment (reachjainrahul) [inline pkg/controllers/inventory/inventory.go:79]: can you follow if err = inventory.vpcStore.Update(discoveredVpc); err == nil { } pattern. Avoid new line for err

Comment (reachjainrahul) [reply pkg/controllers/inventory/inventory.go:79]: lets clean this up in next PR.. Allowing the checkin for now..

**Pull request merged**